src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6716,10 +6722,18 @@ public override BoundNode VisitFieldInfo(BoundFieldInfo node)
         }
 
         public override BoundNode VisitDefaultExpression(BoundDefaultExpression node)
+        {
+            // Can occur in error scenarios
```

Comment:
>// Can occur in error scenarios [](start = 12, length = 31)

It looks like the literal node can survive initial binding too. Is this expected? Shouldn't we always transform it and report an error when necessary?

Reply:
Converted tuple literals currently hold onto the unconverted tuple, and that gets visited by `NullableWalker`. That's one scenario where the bound node for `default` literal survives initial binding.
I'll update the comment in `BoundNodes.xml` to reflect.

---
In reply to: [314117613](https://github.com/dotnet/roslyn/pull/37596#discussion_r314117613) [](ancestors = 314117613)